Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -40,16 +65,18 @@ TIFFImageIO::CanReadFile(const char * file)
   }
 
   // Now check if this is a valid TIFF image
-  TIFFErrorHandler save = TIFFSetErrorHandler(nullptr);
-  int              res = m_InternalImage->Open(file);
-  if (res)
-  {
-    TIFFSetErrorHandler(save);
-    return true;
-  }
-  m_InternalImage->Clean();
-  TIFFSetErrorHandler(save);
-  return false;
+  TIFFErrorHandler savedErrorHandler = TIFFSetErrorHandler(nullptr);
```

Comment:
So the problem with this is that this changes globals in TIFF and non-thread safe way.

The error messages should also go through the ITK messaging system and not straight to stdout/err the way the TIFF error handler does

Starting with libtiff 4.5 introduces error handing that is connected to the TIFF handle ( re-entrant ) and thus not global. 

https://libtiff.gitlab.io/libtiff/functions/TIFFError.html#c.TIFFErrorExtR

Reply:
> So the problem with this is that this changes globals in TIFF and non-thread safe way.

I see ... the current use of `TIFFSetErrorHandler` is already problematic then ?

How about using a Schwartz Counter to ensure the handler is installed on library loading ?